app/models/host_aspect_base.rb

```diff
@@ -0,0 +1,36 @@
+class HostAspectBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  def self.inherited(subclass)
+    super
+    subclass.class_eval do
+      belongs_to_host
+    end
+  end
+
+  def info
+    {}
+  end
+
+  def smart_proxy_ids
+    []
+  end
+
+  def after_clone
+  end
+
+  def template_filter_options(kind)
+    {}
+  end
+
+  def provisioning_template_options(base_options)
+    base_options
+  end
+
+  def self.inherited_attributes(hostgroup, aspect_attributes)
+    aspect_attributes
+  end
+
+  def self.populate_fields_from_facts(host, importer, type, proxy_id)
```

Comment:
It is actually broken in our code: in `Host::Base` the definition of `self.import_host_and_facts` doesn't need a proxy, but in `Host::Managed` it already does. That means that the inheritance doesn't work already (you have to treat Base and Managed differently anyway).
I would like to see this mechanism refactored, but I didn't want to do it inside this PR, for the sake of less complication.